coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala

```diff
@@ -175,6 +175,105 @@ class MultiPartitionPlannerSpec extends AnyFunSpec with Matchers {
+      TimeStepParams(endSeconds, step, endSeconds)
+    )
+
+    val promQlQueryParams = PromQlQueryParams("test{job = \"app\"}", endSeconds, step, endSeconds)
+
+    val execPlan = engine.materialize(lp, QueryContext(origQueryParams = promQlQueryParams,  plannerParams =
+      PlannerParams(processMultiPartition = true)))
+
+    execPlan.printTree()
+
+    execPlan.isInstanceOf[LocalPartitionDistConcatExec] shouldEqual (true)
+    execPlan.children.length shouldEqual 2
+    execPlan.children.head.isInstanceOf[MultiSchemaPartitionsExec] shouldEqual true
+    execPlan.children.head.rangeVectorTransformers.head.isInstanceOf[PeriodicSamplesMapper] shouldEqual true
+
+    val distConcatPlan = execPlan.asInstanceOf[LocalPartitionDistConcatExec]
+
+    val localExec1 = distConcatPlan.children(0).asInstanceOf[MultiSchemaPartitionsExec]
+    val queryParams1 = localExec1.queryContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+    queryParams1.startSecs shouldEqual endSeconds
+    queryParams1.endSecs shouldEqual endSeconds
+    queryParams1.stepSecs shouldEqual step
+    val samplesMapper1 = localExec1.rangeVectorTransformers.head.asInstanceOf[PeriodicSamplesMapper]
+    samplesMapper1.startMs shouldEqual 4200000
+    samplesMapper1.endMs shouldEqual 9600000
+    samplesMapper1.stepMs shouldEqual 10*60*1000
+
+    val localExec2 = distConcatPlan.children(1).asInstanceOf[MultiSchemaPartitionsExec]
+    val queryParams2 = localExec2.queryContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+    queryParams2.startSecs shouldEqual endSeconds
+    queryParams2.endSecs shouldEqual endSeconds
+    queryParams2.stepSecs shouldEqual step
+    val samplesMapper2 = localExec2.rangeVectorTransformers.head.asInstanceOf[PeriodicSamplesMapper]
+    samplesMapper2.startMs shouldEqual 4200000
+    samplesMapper2.endMs shouldEqual 9600000
+    samplesMapper2.stepMs shouldEqual 10*60*1000
+  }
+
+  it ("should generate both PromQlExec and MultiSchemaPartitionsExec for TopLevelSubquery") {
+    val partitionLocationProvider = new PartitionLocationProvider {
+      override def getPartitions(routingKey: Map[String, String], timeRange: TimeRange): List[PartitionAssignment] =
+        List(
+          PartitionAssignment("local", "local-url", TimeRange(timeRange.startMs, timeRange.endMs)),
+          PartitionAssignment("remote", "remote-url", TimeRange(timeRange.startMs, timeRange.endMs))
+        )
+
+      override def getAuthorizedPartitions(timeRange: TimeRange): List[PartitionAssignment] =
+        List(PartitionAssignment("local", "local-url", TimeRange(timeRange.startMs, timeRange.endMs)))
+    }
+
+    val engine = new MultiPartitionPlanner(partitionLocationProvider, localPlanner, "local", dataset, queryConfig)
+    val lp = Parser.queryRangeToLogicalPlan(
+      "test{job = \"app\"}[100m:10m]",
+      TimeStepParams(endSeconds, step, endSeconds)
+    )
+
+    val promQlQueryParams = PromQlQueryParams("test{job = \"app\"}", endSeconds, step, endSeconds)
```

Comment:
Same as above. Query should be 'test{job = \"app\"}[100m:10m]'

Reply:
fixed